darwin/future/core/types/common.py

```diff
@@ -5,14 +5,27 @@
 from darwin.future.data_objects import validators as darwin_validators
 
 
-class TeamSlug(ConstrainedStr):
+class TeamSlug(str):
     """Team slug type"""
 
     min_length = 1
     max_length = 256
 
-    def __new__(self) -> "TeamSlug":
-        return super().__new__(darwin_validators.parse_name(super().__str__))
+    @classmethod
+    def __get_validators__(cls): #type: ignore
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v: str) -> "TeamSlug":
```

Comment:
I think I like this approach better. It's a bit more verbose but shouldn't be too many instances where we have to extend base pydantic objects like this

Reply:
Yeah, I agree.  It's just a bit weird that it's not extended from BaseModel like usual!